lib/npg_qc/autoqc/checks/pulldown_metrics.pm

```diff
@@ -56,28 +57,51 @@ has 'max_java_heap_size' => (
     default => $MAX_JAVA_HEAP_SIZE,
 );
 
-has 'picard_jar_path' => (
+has 'picard_module' => (
     is      => 'ro',
-    isa     => 'NpgCommonResolvedPathJarFile',
-    coerce  => 1,
-    default => $PICARD_JAR_NAME,
+    isa     => 'Str',
+    default => $PICARD_COMMAND,
 );
 
-has 'picard_command' => (
-    is         => 'ro',
-    isa        => 'Str',
+# TODO: This needs to be factored out of here, insert_size and alignment_filter_metrics
+has 'reference' => (
+    is => 'ro',
+    isa => 'Str',
     lazy_build => 1,
 );
 
-sub _build_picard_command {
+sub _build_reference {
     my $self = shift;
-    my $command = $self->java_cmd . sprintf q[ -Xmx%s -jar %s VALIDATION_STRINGENCY=SILENT BAIT_INTERVALS=%s TARGET_INTERVALS=%s INPUT=%s OUTPUT=/dev/stdout],
-        $self->max_java_heap_size,
-        $self->picard_jar_path,
-        $self->bait_intervals_path,
-        $self->target_intervals_path,
-        $self->input_files->[0];
-    return $command;
+    my $ref;
+    try {
+        $ref = pop @{$self->refs()};
+    }
+    catch {
+        $self->result->add_comment(
+            q[Error: binary reference cannot be retrieved; cannot run insert size check. ] . $_
+        );
+    };
+    if (!$ref) {
+        $self->result->add_comment('Failed to retrieve reference');
+    }
+    return $ref;
+}
+
+has 'picard_arguments' => (
+    is => 'ro',
+    isa => 'ArrayRef[Str]',
+    lazy_build => 1,
+);
+
+sub _build_picard_arguments {
+    my $self = shift;
+    return [
```

Comment:
VALIDATION_STRINGENCY=SILENT seems to be lost

Reply:
According to [the manual](https://gatk.broadinstitute.org/hc/en-us/articles/4404607472027-CollectHsMetrics-Picard-#--VALIDATION_STRINGENCY) it seems to only relate to BAM files. It doesn't seem to do any harm to leave it there. Keep or discard?

Reply:
The manual says that it can speed up processing of BAM files, it does not say that it is only relevant to BAM files. GATK tools are very fussy about input data, setting this level of validation used to suppress some errors. We'd better have it, unless it will prove to be inconsistent with the CRAM format.